### ISSUE TITLE
The route `/blog/rss.xml` doesn't get built during the build

### ISSUE THREAD

**Issue by ematipico** [HiDeoo/starlight-blog#107]
### Describe the bug

The route `/blog/rss.xml` isn't built when running `astro build`. It is, however, present in dev

### To Reproduce

Stackblitz repro: https://stackblitz.com/edit/github-t8s75sjf?file=astro.config.mjs,src%2Fcontent%2Fdocs%2Fblog%2Fpost.md&on=stackblitz

### Expected behavior

As documented, it should be built when adding `site`

### How often does this bug happen?

Every time

**Comment by HiDeoo** [comment]
Thanks for the report.

I don't think the linked repro is complete so I tested directly on the Biome website.

First, the code to inject the route is fairly simple:

```js
if (astroConfig.site) {
  injectRoute({
    entrypoint: 'starlight-blog/routes/rss',
    pattern: '/[...prefix]/rss.xml',
    prerender: true,
  })
}
```

The `injectRoute()` function is indeed called as expected in this case.

It looks like commenting the Netlify adapter in the `astro.config.ts` file makes it work as expected and the RSS feeds are generated correctly when running `pnpm build:no-og`:

```sh
λ starlight-blog/routes/rss.xml.ts
  ├─ /blog/rss.xml (+7ms)
  ├─ /fr/blog/rss.xml (+3ms)
  ├─ /ja/blog/rss.xml (+5ms)
  ├─ /zh-cn/blog/rss.xml (+2ms)
  └─ /pt-br/blog/rss.xml (+5ms)
```

Not quite sure what's going on here, will need to investigate further.

**Comment by ematipico** [comment]
That's definitely weird, I'll see if I can remove the adapter for the time being

**Comment by tube2sheet** [comment]
Hello,
I am facing the same issue on my deployment. rss.xml returns 404 error.

I am not proficient enough to say if its because of the Plugin or some other reason. Just wanted to let you know.

**Comment by HiDeoo** [comment]
> Hello, I am facing the same issue on my deployment. rss.xml returns 404 error.
> 
> I am not proficient enough to say if its because of the Plugin or some other reason. Just wanted to let you know.

Which version of Astro are you using? The fix for this issue was released in Astro `5.1.2`. Another option could be that the Astro [`site`](https://docs.astro.build/en/reference/configuration-reference/#site) configuration option in the `astro.config.mjs` file is properly set.

If not, feel free to open a new issue so we can properly track it 👍
